### PR TITLE
ETCM-8833 Run tests from Partner-Chains Repo

### PR DESCRIPTION
**Pull request by ladamesny**
Updates to have nightly, and ci tests run from partner-chains repo.

# Description

Add your description here, if it fixes a particular issue please provide a
[link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword=)
to the issue.

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages.
- [x] New tests are added if needed and existing tests are updated.
- [x] Relevant logging and metrics added
- [x] CI passes. See note on CI.
- [x] Any changes are noted in the `changelog.md` for affected crate
- [x] Self-reviewed the diff

# Note on CI
If your PR is from a fork, the necessary CI jobs won't trigger automatically for security reasons.
You will need to get someone with write privileges.  Please contact IOG Partner Chains developers to do this
for you.

### CODE REVIEW

File: E2E-tests/config/substrate/staging_stack.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
         "ogmios_port": 1337,
         "kupo_host": "staging-preview-services-service.staging-preview.svc.cluster.local",
```
Comment (chrispalaskas) [inline E2E-tests/config/substrate/staging_stack.json:5]: Irrelevant with this PR, just thought we should change the name of the file to staging_preview_stack.json

File: E2E-tests/src/cardano_cli.py [hunk 2]
```diff
@@ -11,7 +11,7 @@ def __init__(self, config: MainChainConfig, cardano_cli: Tool):
         self.run_command = RunnerFactory.get_runner(cardano_cli.ssh, cardano_cli.shell)
 
     def query_tip(self) -> int:
-        cmd = f"{self.cli} query tip {self.network}"
+        cmd = f"{self.cli} latest query tip {self.network}"
```
Comment (chrispalaskas) [inline E2E-tests/src/cardano_cli.py:14]: Why is this here? We don't need it with 10.1.1.0 cli.

File: E2E-tests/src/cardano_cli.py [hunk 3]
```diff
@@ -28,7 +28,7 @@ def get_sync_progress(self) -> float:
         return self.query_tip()["syncProgress"]
 
     def get_utxos(self, addr):
-        cmd = f"{self.cli} query utxo --address {addr} {self.network} --out-file /dev/stdout"
+        cmd = f"{self.cli} latest query utxo --address {addr} {self.network} --out-file /dev/stdout"
```
Comment (chrispalaskas) [inline E2E-tests/src/cardano_cli.py:31]: same.

**Comment by skylar-simoncelli** [comment]
I'm pretty sure you can replace `tools_host` with `binary-host-service.sc.svc.cluster.local` and then this will never need to be updated. Should avoid using IP addressing

**Comment by skylar-simoncelli** [comment]
I'm happy with the structure and the changes to CI, all looks good to me as long as it looks good to @chrispalaskas and @rsporny

File: E2E-tests/src/cardano_cli.py (hunk 2 above)
Comment (ladamesny) [reply E2E-tests/src/cardano_cli.py:14]: It currently breaks without this. I'd like to fix this in a separate PR.

File: E2E-tests/src/cardano_cli.py (hunk 3 above)
Comment (ladamesny) [reply E2E-tests/src/cardano_cli.py:31]: Yes, let's fix in a separate PR.

File: .github/actions/tests/local-environment-tests/action.yml [hunk 4]
```diff
@@ -42,27 +42,12 @@ runs:
...
-        ref: ${{ github.head_ref || github.ref_name }}
-        path: dev/local-environment/configurations/tests/e2e-tests
-    - name: checkout pc-tests master (fallback)
-      if: steps.pc-tests.outcome == 'failure'
-      uses: actions/checkout@v4
-      with:
-        repository: input-output-hk/sidechains-tests
-        token: ${{ env.ACTIONS_PAT }}
-        ref: master
-        path: dev/local-environment/configurations/tests/e2e-tests
```
Comment (rsporny) [inline .github/actions/tests/local-environment-tests/action.yml:61]: @ladamesny have you checked pc-demo? I can't see any changes in `dev/local-environment/configurations/tests/entrypoint.sh` so it may have trouble finding the tests

Comment (rsporny) [reply .github/actions/tests/local-environment-tests/action.yml:61]: apologies, I see the new lines :)

Comment (ladamesny) [reply .github/actions/tests/local-environment-tests/action.yml:61]: yeah I didn't have to change it, because that file still references `./devnet/local-environment/configurations/tests/e2e-tests` folder, I updated the script to copy over our test files from `./E2E-tests/` to `./devnet/local-environment/configurations/tests/e2e-tests`.

File: .github/actions/tests/run-e2e-tests/action.yml [hunk 5]
```diff
@@ -95,22 +95,28 @@ runs:
...
             source venv/bin/activate &&
+            pip install -r requirements.txt &&
             $pytest_cmd
           "
         else
           echo "Running tests locally"
           cd e2e-tests
+          python -m venv venv
           source venv/bin/activate
+          pip install -r requirements.txt
```
Comment (rsporny) [inline .github/actions/tests/run-e2e-tests/action.yml:108]: I think the new lines are redundant, it should be handled in `dev/local-environment/configurations/tests/entrypoint.sh` already

File: E2E-tests/Earthfile [hunk 6]
```diff
@@ -108,5 +110,4 @@ report:
     RUN echo ERROR: Unknown error.
   END
   RUN echo $summary
-  RUN exit $exitcode
```
Comment (rsporny) [inline E2E-tests/Earthfile:111]: why you're removing this?

File: E2E-tests/config/substrate/devnet_nodes.json [hunk 7]
```diff
@@ -63,7 +63,7 @@
                 "rotation_candidate": true,
                 "cardano_payment_addr": "addr_test1vr88zd5ywcdu07hqpnutd8umqyld34e6rrtzgu7m3rua67gg6clcz",
                 "keys_files": {
-                    "cardano_payment_key": "./secrets/substrate/devnet/keys/eve/payment.skey.json.decrypted",
```
Comment (rsporny) [inline E2E-tests/config/substrate/devnet_nodes.json:66]: did we move all secrets on binary-host?

File: E2E-tests/config/substrate/staging_stack.json (hunk 1 above)
Comment (rsporny) [reply E2E-tests/config/substrate/staging_stack.json:5]: staging -> staging_preview

File: .github/actions/tests/local-environment-tests/action.yml [hunk 8]
```diff
@@ -42,27 +42,12 @@ runs:
...
     - name: Deploy local environment with overrides
       run: |
         mv dev/local-environment/configurations/pc-contracts-cli/overrides/partner-chains-cli-artifact dev/local-environment/configurations/pc-contracts-cli/overrides/partner-chains-cli
         mv dev/local-environment/configurations/pc-contracts-cli/overrides/partner-chains-node-artifact dev/local-environment/configurations/pc-contracts-cli/overrides/partner-chains-node
+        mkdir -p dev/local-environment/configurations/tests/e2e-tests
+        cp -r E2E-tests/* dev/local-environment/configurations/tests/e2e-tests
```
[the remark this answers is not in the document]
Comment (ladamesny) [reply .github/actions/tests/local-environment-tests/action.yml:50]: Yeah I tried this in a previous iteration, but got confused, and couldn't get it to work. @skylar-simoncelli pointed out that copying to this folder was the quickest way to make this work. I have it in a separate PR, I'll look at it again and see if I can make it work now.

**Review by chrispalaskas — APPROVED** [review]
Please answer Radek's comments as well.
